### DISCUSSION TITLE
솔루션 사용신청시 판매자의 채널네임(브랜드명) 정보를 얻는 방법이 있을까요?

### DISCUSSION THREAD

**Discussion by billyLee1004** [commerce-api-naver/commerce-solutions#28] [묻고 답하기] [answered]
안녕하세요
이벤트 훅 등록하고 수신 이벤트도 추가하여 판매자가 저희 솔루션 "추가하기"했을 경우
이벤트를 받았습니다.
수신된 body에 accountId,accountUid.. 기타 정보를 가지고 오는 것 까지는 확인 했는데 요기서 판매자의 스토명(채널이름) 을 추가적으로 알고 싶은데 어떻게 해야 하나요?

**Comment by commerce-api-naver** [comment]
안녕하세요. 커머스API 이민석입니다.

### 이벤트 훅에서의 채널명 제공
이벤트 훅에서는 채널명이 별도로 제공되지 않습니다. 신규로 제공하는 방안 역시 가까운 장래에 계획하고 있는 내용이 없습니다.
이벤트 훅은 구독자의 솔루션 구독과 관련한 일련의 이벤트 트리거 발생 시 솔루션 시스템에 주요한 정보를 빠르게 알리려는 목적으로 설계되었습니다.
이러한 설계 목적상 채널명은 부득이하게 제공 정보에서 제외되었습니다.

### 이벤트 훅 내 매핑 필드 활용 팁
만일 운용하시려는 솔루션이 '심사 후 승인'형 솔루션이라면 사용 승인 시 대상 사용자 구독건의 매핑 필드(레퍼런스 정보)를 입력할 수 있습니다.
- 대상 API: [솔루션] 사용 시작 승인 (`PUT /v1/commerce-solutions/subscriptions/approve`)
- 필드명: `accountMappingId` 필드 (Request body에 JSON 형식으로 호출)

사용 시작 승인 시 입력하신 매핑 정보는 이벤트 훅 내 `accountMappingId` 필드로 제공합니다.
솔루션 시스템에서 커머스솔루션 플랫폼의 '사용자 계정 UID(`accountUid`)'가 아닌 별도의 식별 체계로 채널명을 비롯한 사용자 정보를 관리한다면 이를 식별할 수 있는 값을 매핑 필드로 연동할 수도 있습니다.

> [!IMPORTANT]
> 매핑 필드로 연동하는 데이터는 계정 정보 등의 RAW 문자열이 아닌 솔루션 시스템만 알아볼 수 있는 난수 값, 시퀀스 값, 해시 값 등으로 연동하는 것을 권장합니다.

**Comment by billyLee1004** [comment, marked as answer]
넹.  심사후 승인형 솔루션은 아닙니다. 
그럼 이벤트 훅에서 말고 솔루션 구독한 판매자의 스토어명(채널명)를 가지고 오는  api 있을까요? 있으면 어떻게 하면 되나요?

**Reply by commerce-api-naver** [reply]
@billyLee1004 님, 커머스API 이민석입니다.

### JWT를 이용한 채널명 확인
'바로 사용'형 솔루션인 경우 사용자의 JWT 토큰 내 채널 정보를 제공합니다.
- 대상 필드: `channelName`
    - 상세 내용은 관련한 [Wiki 가이드 문서](https://github.com/commerce-api-naver/commerce-solutions/wiki/%5B%EA%B0%80%EC%9D%B4%EB%93%9C%5D-%EC%86%94%EB%A3%A8%EC%85%98-%EC%82%AC%EC%9A%A9%EC%9E%90-JWT-%EA%B5%AC%EC%84%B1%EC%9A%94%EC%86%8C-%EB%B0%8F-%EC%97%B0%EB%8F%99-%EC%A3%BC%EC%9D%98%EC%82%AC%ED%95%AD) 참고

### 커머스API를 이용한 채널명 확인
①구독자의 솔루션의 정상 구독 상태에서 ②대상 솔루션이 [판매자정보] API 그룹 권한이 있다면 - 아래 API로 확인이 가능합니다.
- 대상 API: [판매자정보] 계정으로 채널 정보 조회 (`GET /v1/seller/channels`)
    - Response 내 `[n].name` 필드로 제공
